src/test/java/seedu/address/logic/commands/CheckInCommandTest.java

```diff
@@ -0,0 +1,138 @@
+package seedu.address.logic.commands;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+
+import org.junit.Test;
+
+import seedu.address.logic.CommandHistory;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.room.RoomNumber;
+import seedu.address.testutil.TypicalConcierge;
+import seedu.address.testutil.TypicalRoomNumbers;
+
+/**
+ * Contains integration tests (interaction with the Model, UndoCommand and RedoCommand) and unit tests for
+ * {@code CheckInCommand}.
+ */
+public class CheckInCommandTest {
+
+    private Model model = new ModelManager(TypicalConcierge.getTypicalConciergeWithRoomBookings(), new UserPrefs());
+    private CommandHistory commandHistory = new CommandHistory();
+
+    @Test
+    public void execute_invalidCheckinExpiredBooking_throwsExpiredBookingException() {
+        RoomNumber roomNumberToCheckIn = TypicalRoomNumbers.ROOM_NUMBER_001;
+        CheckInCommand checkInCommand = new CheckInCommand(roomNumberToCheckIn);
```

Comment:
Short answer: No, because doing so changes the model's VersionedAddressBook's list of states. That is why I removed the @Before initializer that was present before this PR. There is a way to circumvent that, but it opens up more possibilities for unintended behavior that way. If you want to know the full details, ask me again. 

See how Model is constructed. It uses the bookings in TypicalBookings.

No, it will not. Did you see how Model is constructed here, and not from TestApp?